packages/flutter/lib/src/painting/colors.dart

```diff
@@ -451,7 +451,7 @@ class ColorSwatch<T> extends Color {
       return false;
     return super == other
         && other is ColorSwatch<T>
-        && other._swatch == _swatch;
+        && MapEquality<T, Color>().equals(other._swatch, _swatch);
```

Comment:
Please use https://master-api.flutter.dev/flutter/foundation/mapEquals.html from the foundation library instead of importing package:collection

Reply:
Okay thanks for telling. :)